src/Imghoard/ImghoardClient.cs

```diff
@@ -91,38 +135,75 @@ public async Task<Uri> PostImageAsync(Stream image, params string[] Tags)
                 await image.CopyToAsync(mStream);
                 bytes = mStream.ToArray();
             }
+            image.Position = 0;
 
-            var imgd = IsSupported(bytes);
+            return await PostImageAsync(bytes, Tags);
+        }
 
-            if (!imgd.Item1)
+        /// <summary>
+        /// Posts a new image to the Imghoard instance
+        /// </summary>
+        /// <param name="bytes">The raw bytes of the image to upload</param>
+        /// <param name="Tags">The tags of the image being uploaded</param>
+        /// <returns>The url of the uploaded image or null on failure</returns>
+        public async Task<string> PostImageAsync(Memory<byte> bytes, params string[] Tags)
+        {
+            if(bytes.Length >= 1000000 && !config.Experimental)
             {
-                throw new NotSupportedException(
-                    "You have given an incorrect image format, currently supported formats are: png, jpeg, gif");
+                throw new NotSupportedException("In order to upload images larger than 1MB you need to enable experimental features in the config");
             }
 
-            var b64 = Convert.ToBase64String(bytes);
-            image.Position = 0;
+            (bool supported, string prefix) = IsSupported(bytes.Span.ToArray());
```

Comment:
This copies the array twice. I would recommend for the future to move `IsSupported` to a span/memory as well.